src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AdminUserService.java

```diff
@@ -17,11 +24,18 @@ public class AdminUserService {
     @Value("${etcos.system.password}")
     private String systemUserPassword;
 
+    @Cacheable("adminUserToken")
     public String getAdminUserToken() {
         return String.join(" ", BEARER, userIdamService.getAccessToken(systemUserName, systemUserPassword));
     }
 
-    public UserDetails getUserDetails(String userId) {
-        return userIdamService.getUserDetailsById(getAdminUserToken(), userId);
+    @CacheEvict(value = "adminUserToken", allEntries = true)
+    @Scheduled(fixedRateString = "${caching.adminUserService}")
+    public void emptyAdminUserToken() {
+        log.info("emptying adminUserToken cache");
+    }
+
+    public UserDetails getUserDetails(String accessToken, String userId) {

```

Comment:
We can also cache this method, or add caching to userIdamService.

Reply:
Not caching this as it's not something I think we want to cache. UserDetails can change so say if we have cached an ID and the user details change, we could be pulling outdated information. Plus the usage of it isn't as frequent as the admin token